pkg/intermediate/aggregate_test.go

```diff
@@ -713,45 +742,34 @@ func TestGetRecords(t *testing.T) {
 	recordIPv6Dst := createDataMsgForDst(t, true, false, false, false, false).GetSet().GetRecords()[0]
 
 	records := []entities.Record{recordIPv4Src, recordIPv4Dst, recordIPv6Src, recordIPv6Dst}
-
-	flowKeyIPv4, _, _ := getFlowKeyFromRecord(recordIPv4Src)
-
 	for _, record := range records {
 		flowKey, isIPv4, _ := getFlowKeyFromRecord(record)
 		err := ap.addOrUpdateRecordInMap(flowKey, record, isIPv4)
 		assert.NoError(t, err)
 	}
+
+	flowKeyIPv4, _, _ := getFlowKeyFromRecord(recordIPv4Src)
 	recordStrings := ap.GetRecords(flowKeyIPv4)
 	assert.Equalf(t, 1, len(recordStrings), "Number of IPv4 records string is incorrect, expected 1 got %d", len(recordStrings))
-	assert.Contains(t, recordStrings[0], "    sourceTransportPort: 1234 \n")
-	assert.Contains(t, recordStrings[0], "    destinationTransportPort: 5678 \n")
-	assert.Contains(t, recordStrings[0], "    protocolIdentifier: 6 \n")
-	assert.Contains(t, recordStrings[0], "    sourcePodName: pod1 \n")
-	assert.Contains(t, recordStrings[0], "    destinationPodName: pod2 \n")
-	assert.Contains(t, recordStrings[0], "    sourceIPv4Address: 10.0.0.1 \n")
-	assert.Contains(t, recordStrings[0], "    destinationIPv4Address: 10.0.0.2 \n")
-	assert.Contains(t, recordStrings[0], "    destinationClusterIPv4: 192.168.0.1 \n")
-	assert.Contains(t, recordStrings[0], "    destinationServicePort: 4739 \n")
-	assert.Contains(t, recordStrings[0], "    flowEndSeconds: 1 \n")
-	assert.Contains(t, recordStrings[0], "    flowType: 2 \n")
-	assert.Contains(t, recordStrings[0], "    flowEndReason: 2 \n")
-	assert.Contains(t, recordStrings[0], "    tcpState: ESTABLISHED \n")
-	assert.Contains(t, recordStrings[0], "    ingressNetworkPolicyRuleAction: 0 \n")
-	assert.Contains(t, recordStrings[0], "    egressNetworkPolicyRuleAction: 0 \n")
-	assert.Contains(t, recordStrings[0], "    ingressNetworkPolicyRulePriority: 50000 \n")
-	assert.Contains(t, recordStrings[0], "    packetTotalCount: 500 \n")
-	assert.Contains(t, recordStrings[0], "    packetDeltaCount: 0 \n")
-	assert.Contains(t, recordStrings[0], "    reversePacketTotalCount: 500 \n")
-	assert.Contains(t, recordStrings[0], "    reversePacketDeltaCount: 0 \n")
+	assertRecordString(t, recordStrings[0], false)
 
 	partialFlowKeyIPv6 := &FlowKey{
 		SourceAddress: "2001:0:3238:dfe1:63::fefb",
 	}
 	recordStrings = ap.GetRecords(partialFlowKeyIPv6)
 	assert.Equalf(t, 1, len(recordStrings), "Number of IPv6 records string is incorrect, expected 1 got %d", len(recordStrings))
-	assert.Contains(t, recordStrings[0], "    sourceIPv6Address: 2001:0:3238:dfe1:63::fefb \n")
-	assert.Contains(t, recordStrings[0], "    destinationIPv6Address: 2001:0:3238:dfe1:63::fefc \n")
-	assert.Contains(t, recordStrings[0], "    destinationClusterIPv6: 2001:0:3238:bbbb:63::aaaa \n")
+	assertRecordString(t, recordStrings[0], true)
+
+	recordStrings = ap.GetRecords(nil)
```

Comment:
How about initializing all three testcases with a struct and use t.Run? https://gobyexample.com/testing
It will be easy to follow all testcases.

Reply:
Updated